### PR TITLE
Implement tooltips in multiple pages and components

### PR DESCRIPTION
**Pull request by rating89us**

### CODE REVIEW

**Comment by selsta** [comment]
Really nice, thanks :) Will review soon.

**Comment by rating89us** [comment]
ok, it's ready for review

**Comment by selsta** [comment]
We could also add a delay + opacity effect.

```patch
diff --git a/components/Tooltip.qml b/components/Tooltip.qml
index f2e71def..b60b007e 100644
--- a/components/Tooltip.qml
+++ b/components/Tooltip.qml
@@ -67,7 +67,7 @@ Rectangle {
         }
     }
 
-    Popup {
+    ToolTip {
         id: popup
 
         background: Rectangle {
@@ -84,7 +84,14 @@ Rectangle {
         y: tooltipBottom
             ? (tooltipIconVisible ? icon.y + height : parent.y + parent.height + 2)
             : (tooltipIconVisible ? icon.y - height : parent.y - tooltip.height - 20)
+        enter: Transition {
+            NumberAnimation { property: "opacity"; from: 0.0; to: 1.0; duration: 150 }
+        }
 
+        exit: Transition {
+            NumberAnimation { property: "opacity"; from: 1.0; to: 0.0; duration: 150 }
+        }
+        delay: 200
         RowLayout {
             Text {
                 id: tooltip
```

Note that this patch breaks Transfer -> Advanced Options tooltips, I did not figure out why yet.

**Comment by handwavingdrivendevelopment** [comment]
Will these Tooltips also work with taps on a tablet/mobile device or do they only work on mouseenter/leave events?

**Comment by rating89us** [comment]
> Will these Tooltips also work with taps on a tablet/mobile device 

Currently they are working if you tap and hold on Android, but I don't think they are really useful on mobile, because sometimes you can't read the tooltip since your finger stays on top of it

**Comment by rating89us** [comment]
> We could also add a delay + opacity effect.

Done. 

> Note that this patch breaks Transfer -> Advanced Options tooltips, I did not figure out why yet.
Actually all tooltips where affected. It was just a matter of setting a height to the ToolTip component.

File: components/AdvancedOptionsItem.qml [hunk 1]
```diff
@@ -22,6 +22,7 @@ RowLayout {
         MoneroComponents.Label {
             id: title
             fontSize: 14
+            tooltipIconVisible: true
```
Comment (selsta) [inline components/AdvancedOptionsItem.qml:25]: What is this used for? I wasn't able to figure it out.

File: pages/Transfer.qml [hunk 2]
```diff
@@ -483,12 +486,16 @@ Rectangle {
                                 horizontalAlignment: Text.AlignHCenter
                                 opacity: mouseArea.containsMouse ? 1 : 0.85
                                 text: recipientModel.count == 1 ? FontAwesome.infinity : FontAwesome.times
+                                tooltip: recipientModel.count == 1 ? qsTr("Send all unlocked balance of this account") :  qsTr("Remove recipient")  + translationManager.emptyString
```
Comment (selsta) [inline pages/Transfer.qml:489]:
```suggestion
                                tooltip: recipientModel.count == 1 ? qsTr("Send all unlocked balance of this account") : qsTr("Remove recipient")  + translationManager.emptyString
```

File: components/AdvancedOptionsItem.qml (hunk 1 above)
Comment (rating89us) [reply components/AdvancedOptionsItem.qml:25]: When tooltipIconVisible = true we display the question mark icon that when hovered turns tooltip visible:
image

When tooltipIconVisible = false (default), we display only the tooltip box.

Comment (selsta) [reply components/AdvancedOptionsItem.qml:25]: Ah I looked on the wrong page :)

**Comment by selsta** [comment]
Please squash and then I can approve it.

**Comment by selsta** [comment]
Where is the checkbox tooltip used?

**Comment by rating89us** [comment]
I'm going to use it in WizardCreateDevice1.qml of PR #3508. Decided to include here in this PR because it's implementing tooltips in multiple components.

**Comment by selsta** [comment]
ok please squash

**Review by selsta — APPROVED** [review]

**Pull request merged**